lib/resty/acme/client.lua

```diff
@@ -301,7 +305,12 @@ function _M:post(url, payload, headers)
     body = json.decode(resp.body)
   elseif resp.headers['Content-Type']:sub(1, 24) == "application/problem+json" then
     body = json.decode(resp.body)
-    return nil, nil, body.detail or body.type
+    if body.type == 'urn:ietf:params:acme:error:badNonce' then
+      log(ngx_DEBUG, "bad nonce: recoverable error, retrying")
+      return self:post(url, payload, headers, resp.headers["Replay-Nonce"])
+    else
```

Comment:
Although RFC says `client SHOULD retry on receiving such response`, i'd suggest to limit the number of retries to avoid bad server implementations. for start we can simply bail out on `badNonce` error and `nonce` is already set (meaning we are in a retry already). how do you think?

Reply:
Sounds good